Python/fileutils.c

```diff
@@ -1750,7 +1750,16 @@ _Py_read(int fd, void *buf, size_t count)
         Py_BEGIN_ALLOW_THREADS
         errno = 0;
 #ifdef MS_WINDOWS
+        _doserrno = 0;
         n = read(fd, buf, (int)count);
+        int winerr = _doserrno;
+        if (n < 0 && errno == EINVAL) {
+                if (winerr == ERROR_NO_DATA) {
+                        /* read() on a non-blocking empty pipe
+                           returns EINVAL on Windows */
+                        errno = EAGAIN;
+                }
+        }
```

Comment:
Per PEP 7, a comment should go before the code that it describes; the C99 `//` comment style is allowed; and indents should be 4 spaces. Also, I think it would be better to clarify the context in the comment instead of adding the redundant variable `winerr`.

```suggestion
        // read() on a non-blocking empty pipe fails with EINVAL, which is
        // mapped from the Windows error code ERROR_NO_DATA.
        if (n < 0 && errno == EINVAL) {
            if (_doserrno == ERROR_NO_DATA) {
                errno = EAGAIN;
            }
        }
```